x-pack/platform/packages/shared/onechat/onechat-server/attachments/type_definition.ts

```diff
@@ -6,16 +6,56 @@
  */
 
 import type { MaybePromise } from '@kbn/utility-types';
+import type { Attachment } from '@kbn/onechat-common/attachments';
 
-export type AttachmentParseResult<TResult = unknown> =
-  | { valid: true; data: TResult }
+/**
+ * Server-side definition of an attachment type.
+ */
+export interface AttachmentTypeDefinition<TType extends string = string, TContent = unknown> {
+  /**
+   * Unique identifier for the attachment type to register.
+   */
+  id: TType;
+  /**
+   * validation function, which will be called when the attachment is added to the conversation.
+   */
+  validate: (input: unknown) => MaybePromise<AttachmentValidationResult<TContent>>;
+  /**
+   * format the attachment to presented to the LLM
+   */
+  format: (attachment: Attachment<TType, TContent>) => MaybePromise<AgentFormattedAttachment>;
+  /**
+   * should return the list of tools which should be exposed to the agent
+   * when attachments of that type are present in the conversation.
+   */
+  getTools?: () => string[];
+  /**
+   * can be used to provide a description of the attachment type,
+   * which will be exposed to the agent when attachments of that type
+   * are present in the conversation.
+   */
+  getAgentDescription?: () => string;
+}
+
+/**
+ * Return type for attachment's validation handlers.
+ * Refer to {@link InlineAttachmentTypeDefinition.validate}
+ */
+export type AttachmentValidationResult<TValidatedData = unknown> =
+  /** valid attachment */
+  | { valid: true; data: TValidatedData }
+  /** validation failed */
```

Comment:
These comments are helpful, maybe you can consider adding some here if you don't break the types down